tests/test_unittest/test_exec_info.py

```diff
@@ -0,0 +1,272 @@
+                -4 * in_phi[0, 0, 0]  # type: ignore
+                + in_phi[-1, 0, 0]  # type: ignore
+                + in_phi[1, 0, 0]  # type: ignore
+                + in_phi[0, -1, 0]  # type: ignore
+                + in_phi[0, 1, 0]  # type: ignore
+            )
+            lap2 = (  # noqa
+                -4 * lap1[0, 0, 0]
+                + lap1[-1, 0, 0]
+                + lap1[1, 0, 0]
+                + lap1[0, -1, 0]
+                + lap1[0, 1, 0]
+            )
+            flux_x = lap2[1, 0, 0] - lap2[0, 0, 0]  # noqa
+            flux_y = lap2[0, 1, 0] - lap2[0, 0, 0]  # noqa
+            out_phi = in_phi + alpha * (  # noqa
+                flux_x[0, 0, 0] - flux_x[-1, 0, 0] + flux_y[0, 0, 0] - flux_y[0, -1, 0]
+            )
+
+    def compile_stencils(self, backend):
+        self.advection = gtscript.stencil(backend=backend, definition=self.advection_def)
+        self.diffusion = gtscript.stencil(backend=backend, definition=self.diffusion_def)
+
+    def init_fields(self, data, backend):
+        self.nx = data.draw(hyp_st.integers(min_value=7, max_value=32), label="nx")
+        self.ny = data.draw(hyp_st.integers(min_value=7, max_value=32), label="ny")
+        self.nz = data.draw(hyp_st.integers(min_value=1, max_value=32), label="nz")
+        shape = (self.nx, self.ny, self.nz)
+
+        self.in_phi = gt_storage.from_array(
+            data.draw(st_arrays(dtype=float, shape=shape)),
+            backend=backend,
+            default_origin=(0, 0, 0),
+            dtype=float,
+        )
+        self.in_u = gt_storage.from_array(
+            data.draw(st_arrays(dtype=float, shape=shape)),
+            backend=backend,
+            default_origin=(0, 0, 0),
+            dtype=float,
+        )
+        self.in_v = gt_storage.from_array(
+            data.draw(st_arrays(dtype=float, shape=shape)),
+            backend=backend,
+            default_origin=(0, 0, 0),
+            dtype=float,
+        )
+        self.tmp_phi = gt_storage.from_array(
+            data.draw(st_arrays(dtype=float, shape=shape)),
+            backend=backend,
+            default_origin=(1, 1, 0),
+            dtype=float,
+        )
+        self.out_phi = gt_storage.from_array(
+            data.draw(st_arrays(dtype=float, shape=shape)),
+            backend=backend,
+            default_origin=(3, 3, 0),
+            dtype=float,
+        )
+        self.alpha = 1 / 32
+
+    def subtest_exec_info(self, exec_info):
+        assert "call_start_time" in exec_info
+        assert "call_end_time" in exec_info
+        assert np.greater(exec_info["call_end_time"], exec_info["call_start_time"])
+
+        assert "run_start_time" in exec_info
+        assert np.greater(exec_info["run_start_time"], exec_info["call_start_time"])
+        assert "run_end_time" in exec_info
+        assert np.greater(exec_info["run_end_time"], exec_info["run_start_time"])
+        assert np.greater(exec_info["call_end_time"], exec_info["run_end_time"])
```

Comment:
Just a question: why is `np.greater` used here? Isn't this just a regular comparison between scalar values?

Reply:
Yeah... I was believing that `np.greater` used some tolerance to account for rounding errors, but it doesn't actually. I will drop it.